src/storage/disk_manager.c

```diff
@@ -4680,19 +4679,20 @@ disk_unreserve_ordered_sectors_without_csect (THREAD_ENTRY * thread_p, DB_VOLPUR
   context.purpose = purpose;
 
   /* note: vsids are ordered */
-  for (start_index = 0; start_index < nsects; start_index = end_index)
+  anchor = 0;
+  for (index = 0; index < nsects - 1; index++)
     {
-      assert (volid < vsids[start_index].volid);
-      volid = vsids[start_index].volid;
-      for (end_index = start_index + 1; end_index < nsects && vsids[end_index].volid == volid; end_index++)
+      assert (volid < vsids[index].volid);
+      if (volid > vsids[index].volid || index == nsects - 2)
```

Comment:
What if the last one, `nsects -1` has a different volid than `nsects-2`? Doesn't it skip it?

Reply:
Thank you for the advices.
I will change `nsects -2` to `nsects - 1`, and also assert on 4695 to `assert(anchor == nsects - 1)`.

Reply:
The volid is initialized to the NULL_VOLID. And it is never assigned before the if condition. So, the condition of this if statement is always false before the index is equal to the nsects - 2.